site/_posts/2020-12-23-auth-facade-pattern.md

```diff
@@ -0,0 +1,91 @@
+
+Since auth is a necessary part of your application, but not a differntiator, you actually want an auth system that fades into the background. Both in the literal sense, where the UX isn't recognizable as anything separate from your application, and the figurative sense, where your engineers don't need to spend any time worrying about it or maintaining it.
+
+You also want a friendly license for embedding the auth system into your code. I am not a lawyer, but ensure you understand the licensing ramifications of any open source code you include in your app. Some open source projects offer dual licensing, which may be a viable option.
+
+## APIs and data model
+
+If the lowest common denominator of authentication and authorization information works for your needs, following the OIDC and SAML standards may be enough. But often you'll want more information about a user than these standards can provide. You may want to store custom domain specific user attributes, for example.
+
+Any auth facade solution must provide a superior data model when compared to the standards in order to be worthwhile to implement. 
+
+## Documentation
+
+In talking to customers, this surprised me. While documentation is often useful, why is it so important for an auth facade? 
+
+The answer? Because it lets your team offload configuration of the connection between the auth facade and the upstream systems to people who know and maintain those systems. These employees of your clients won't be familiar with your application. In fact, they may resent the additional work that your application's installation requires of them. Making the configuration of the auth facade as well documented and simple as possible is going to make it easier to get your application working.
+
+This leads you with two choices. Maintain such documentation yourself, or rely on the auth provider to be a source of great docs.
+
+## How can you implement the auth facade with FusionAuth
+
+As implied above, many customer have found FusionAuth to be a great auth facade solution. It runs anywhere, integrates with both SAML and OIDC, provides a single, rich interface for your developers, and is well documented.
+
+To learn more or begin implementing, review the [identity provider](/docs/v1/tech/identity-providers/) and [connector](/docs/v1/tech/connectors/) documentation to make sure that your enterprise's target identity providers are supported. [Contact us](/contact/) if they aren't; we'd love to learn more. You can also read [case](/resources/dealcloser-case-study.pdf) [studies](/resources/unsupervised-case-study.pdf) of customers using FusionAuth as an auth facade.
+
+If FusionAuth fits your needs, download FusionAuth it and get going. If you are interested in packaging it as part of your application, please contact us to discuss a resellers agreement.
```

Comment:
If FusionAuth fits your needs, [download FusionAuth](/download/) **_and get it going_**. If you are interested in packaging it as part of your application, please [contact us](/contact/) to discuss a resellers agreement.